internal/extproc/translator/openai_awsbedrock.go

```diff
@@ -152,113 +180,194 @@ func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIToolsToBedrockToolC
 	return nil
 }
 
-// OpenAIMessageToBedrockMessage converts openai ChatCompletion messages to aws bedrock messages
-func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) OpenAIMessageToBedrockMessage(openAIReq *openai.ChatCompletionRequest,
+// openAIMessageToBedrockMessageRoleUser converts openai user role message
+// returns a list of bedrock message and error
+// the tool content is appended to the list of bedrock message list if tool_call is in openai message
+func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIMessageToBedrockMessageRoleUser(
+	openAiMessage *openai.ChatCompletionMessageParamUnion,
+) ([]*awsbedrock.Message, error) {
+	message := openAiMessage.Value.(openai.ChatCompletionUserMessageParam)
+	bedrockMessages := make([]*awsbedrock.Message, 0)
+
+	if _, ok := message.Content.Value.(string); ok {
+		bedrockMessages = append(bedrockMessages, &awsbedrock.Message{
+			Role: openAiMessage.Type,
+			Content: []*awsbedrock.ContentBlock{
+				{Text: ptr.To(message.Content.Value.(string))},
+			},
+		})
+	} else if contents, ok := message.Content.Value.([]openai.ChatCompletionContentPartUserUnionParam); ok {
+		chatMessage := &awsbedrock.Message{Role: openAiMessage.Type}
+		chatMessage.Content = make([]*awsbedrock.ContentBlock, 0, len(contents))
```

Comment:
quick q: why does this need to be a pointer? values slice can be used right?

Reply:
I think this is following the code in the aws sdk, for large content block it might be more efficient to use pointer, otherwise it needs to allocate a large chunk of continuous memory.

Reply:
well that's only true if that's the scale of 100MB or 1Gb, but i don't believe that's the case. i would say this is a bad code of AWS

Reply:
well that's generally untrue. if the size is that big then allocation per item gets a real bottleneck (believe me i wrote a native compiler in wazero project and this is real one) vs if the size is small allocation of the entire contiguous region is much faster and less GC pressure. i would say this is a bad code of AWS and I would suggest not follow their code and fix it (can do this later).